apps/coscrad-frontend/src/components/higher-order-components/categorizable-page-layout.tsx

```diff
@@ -0,0 +1,174 @@
+import { AggregateCompositeIdentifier } from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    Close as CloseIcon,
+    ExpandLess as ExpandLessIcon,
+    ExpandMore as ExpandMoreIcon,
+    Hub as HubIcon,
+    TextSnippet as TextSnippetIcon,
+} from '@mui/icons-material';
+import {
+    Box,
+    Divider,
+    Drawer,
+    IconButton,
+    Stack,
+    Tooltip,
+    Typography,
+    styled,
+} from '@mui/material';
+import { ReactNode, useState } from 'react';
+import { CommandPanel } from '../commands';
+
+interface CategorizablePageLayoutProps {
+    compositeIdentifier: AggregateCompositeIdentifier;
+    selfNotesList: JSX.Element;
+    connectedResourcesList: JSX.Element;
+    commandPanel?: JSX.Element;
+    children: ReactNode;
+}
+
+const DrawerHeader = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'flex-start',
+    margin: theme.spacing(2, 4, 0, 4),
+    justifyContent: 'space-between',
+}));
+
+const DrawerContentStack = styled(Stack)(({ theme }) => ({
+    margin: theme.spacing(0, 4, 0, 4),
+    overflow: 'scroll',
+}));
+
+type Anchor = 'top' | 'left' | 'bottom' | 'right';
+
+export const CategorizablePageLayout = ({
+    compositeIdentifier: { type: resourceType, id },
+    selfNotesList,
+    connectedResourcesList,
+    commandPanel,
+    children,
+}: CategorizablePageLayoutProps) => {
+    const [drawerState, setDrawerState] = useState({
+        bottom: false,
+        right: false,
+    });
+    const [isBottomDrawerExpanded, setIsBottomDrawerExpanded] = useState(false);
+
+    const toggleDrawer = (anchor: Anchor, isOpen: boolean) => {
+        setDrawerState({ ...drawerState, [anchor]: isOpen });
+    };
+
+    return (
+        <>
+            {children}
+            <Box sx={{ textAlign: 'right', pr: 8, mb: 8 }}>
+                <Tooltip title="Open Notes Panel">
+                    <IconButton
+                        data-testid="open-notes-panel-button"
+                        onClick={() => {
+                            toggleDrawer('bottom', !drawerState['bottom']);
+                        }}
+                    >
+                        <TextSnippetIcon />
+                    </IconButton>
+                </Tooltip>
+                <Tooltip title="Open Connected Resources Panel">
+                    <IconButton
+                        data-testid="open-connected-resource-panel-button"
+                        onClick={() => {
+                            toggleDrawer('right', !drawerState['right']);
```

Comment:
Do you want to allow all drawers to be open at once? If not, you probably want to have a more concise way to represent \ update state.

Reply:
It may be overkill here, but useReducer would keep things super clear. I find the state management here hard to follow.